code/modules/antagonists/heretic/magic/aggressive_spread.dm

```diff
@@ -16,6 +16,23 @@
 
 	aoe_radius = 2
 
+/datum/action/cooldown/spell/aoe/rust_conversion/before_cast(atom/cast_on)
+	. = ..()
+	if(. & SPELL_CANCEL_CAST)
+		return
+
+	return SPELL_NO_IMMEDIATE_COOLDOWN
+
+/datum/action/cooldown/spell/aoe/rust_conversion/after_cast(atom/cast_on)
+	. = ..()
+	var/datum/status_effect/heretic_passive/rust/rust_passive = owner.has_status_effect(/datum/status_effect/heretic_passive/rust)
+	if(!rust_passive)
+		StartCooldown(cooldown_time)
+		return
+
+	var/new_cooldown = 35 SECONDS - (rust_passive.passive_level * 5 SECONDS)
+	StartCooldown(new_cooldown)
```

Comment:
My gut `/datum/status_effect/heretic_passive/rust` should just edit `cooldown_time` directly rather than snowflake it like this - if possible

Reply:
I think spells should handle their own cooldowns.
Yeah I can modify it in the rust passive but I think it's more clear when code diving that it's affected here, rather than being silently modified